moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp

```diff
@@ -413,6 +417,11 @@ void PlanningSceneMonitor::scenePublishingThread()
       planning_scene_publisher_.publish(msg);
       if (is_full)
         ROS_DEBUG_NAMED(LOGNAME, "Published full planning scene: '%s'", msg.name.c_str());
+      if(publish_planning_scene_frequency_updated_) {
```

Comment:
Why would you use two variables here instead of a simple atomic double?
Do you think it's necessary to log the Rate "non-"change if the frequency did not change?

Also, `if(flag){ ...; flag= false; }` breaks the use of an atomic. I would have expected [`compare_exchange_strong`](https://en.cppreference.com/w/cpp/atomic/atomic/compare_exchange) or something similar for an atomic.